---
title: "Sunil Ex17 Phospho differential expression & KSEA"
author: "Camilo Posso"
date: "06/01/2021"
output: 
  html_document:
    code_folding: hide
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, include=FALSE}
library(amlresistancenetworks)
library(kableExtra)
library(dplyr)
library(readr)
source("../../Util/synapseUtil.R")

# corrected for loading mass and plexID, long time and subtle
p.site.corrected <- querySynapseTable("syn25706631")

phos.dat <- p.site.corrected
phos.dat$Peptide <- sub("^.*@(.*)$", "\\1", phos.dat$Accession)
phos.dat$site <- sub("^(.*)@.*$", "\\1", phos.dat$Accession)
phos.dat$Gene <- sub("(.*)-.*$", "\\1", phos.dat$site)

# from here we can select just the metadata of interest
metadata.columns = c('Sample','BeatAML Patient ID','Plex','Loading Mass', 'Description')
summary <- phos.dat%>%
  select(metadata.columns)%>%
  distinct()
summary$Description <- as.character(summary$Description)
summary$Type <- case_when(summary$`BeatAML Patient ID` == "Healthy Donor Stroma" ~ "Healthy Donor Stroma",
                          summary$`BeatAML Patient ID` == "Cell line" ~ "Cell line")
summary[is.na(summary$Type), "Type"] <- "Treated"

## Elie has corrected the time period annotation. Adding a pre-treatment, early, and late period.
correct.time.annotation <- as.data.frame(read_csv("../Updated time period annotation.csv"))
summary$Period <- correct.time.annotation[summary$Sample, "Classification"]
summary$Period <- case_when(grepl("early", summary$Period) ~ "Early",
                            grepl("Late", summary$Period) ~ "Late",
                            grepl("pre-treatment", summary$Period) ~ "Pre-Treatment")
rownames(summary) <- as.numeric(summary$Sample)
```

### Get dataset

We access synapse to load the data in long form. We use data which has been corrected for both
Loading mass effect, as well as plex effect. Below we see a metadata table for this experiment.

```{r pressure, echo=FALSE}
show <- summary %>%
  select(-`BeatAML Patient ID`)
kbl(show) %>%
  kable_paper("hover", full_width = F)

```

In this report, we collect the results of rank based GSEA, as well as differential expression. We perform
5 comparisons overall. First, we compare the Healthy group to the Treated group. Now, as we can see,
the Treated group itself consists of 3 subgroups determined by the Period of treatment, namely Pre-Treatment,
Early and Late, so we compare the Healthy group to each of these 3 subgroups as well. Finally, we compare the
Pre-Treatment group directly to the Early group.

```{r prepare groups, limma, and early vs late ksea, include=FALSE}
##select the samples of interest
healthy <- subset(summary, Type == 'Healthy Donor Stroma')%>%
  select(Sample)%>%
  unique()

treated.late <- subset(summary, Period == "Late")%>%
  select(Sample)%>%
  unique()

treated.early <- subset(summary, Period == "Early")%>%
  select(Sample)%>%
  unique()

treated.pre <- subset(summary, Period == "Pre-Treatment") %>%
  select(Sample) %>%
  unique()

treated <- subset(summary, Type == 'Treated')%>%
  select(Sample)%>%
  unique()

##then we spread the proteomics data into a matrix
phos.mat <- phos.dat%>%
  select(LogRatio, Sample, site)%>%
  tidyr::pivot_wider(values_from='LogRatio',names_from='Sample',
                      values_fn=list(LogRatio=mean),values_fill=0.0)%>%
  tibble::column_to_rownames('site')

gene.to.site<-dplyr::select(phos.dat,Gene,site,Peptide)%>%distinct()%>%
    dplyr::mutate(residue=stringr::str_replace(site,paste0(Gene,'-'),''))%>%
    dplyr::mutate(residue=stringr::str_replace_all(residue,"([STY])", ";\\1"))%>%
    dplyr::mutate(residue=stringr::str_replace(residue,"^;", ""))%>%
    dplyr::mutate(residue=stringr::str_replace_all(residue,"([sty])", ""))

## LogFC computed as treated - healthy.
phos.diffex.treated <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, treated$Sample)

## LogFC computed as pre-treatment - healthy
phos.diffex.pre <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, treated.pre$Sample)

## LogFC computed as early - healthy
phos.diffex.early <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, treated.early$Sample)

## LogFC computed as late - healthy
phos.diffex.late <- limmaTwoFactorDEAnalysis(phos.mat, healthy$Sample, treated.late$Sample)

## LogFC computed as early - pre-treatment
phos.diffex.earlyvspre <- limmaTwoFactorDEAnalysis(phos.mat, treated.pre$Sample, treated.early$Sample)

```


```{r computing KSEA, eval=FALSE, include=FALSE}
## Run this to make the pictures. We use PhosphoSite Plus + NetworKIN
set.seed(117)

ksea.treated <- phos.diffex.treated %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Treated", ksea_FDR = 0.05)

ksea.pre <- phos.diffex.pre %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Pre-Treatment", ksea_FDR = 0.05)

ksea.early <- phos.diffex.early %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Treated, early-period", ksea_FDR = 0.05)

ksea.late <- phos.diffex.late %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Healthy-vs-Treated, late-period", ksea_FDR = 0.05)

ksea.earlyvspre <- phos.diffex.earlyvspre %>%
  tibble::rownames_to_column('site') %>%
  left_join(gene.to.site) %>%
  dplyr::select(Gene,Peptide,residue,value='logFC',p_adj='adj.P.Val') %>%
  computeKSEA(prefix = "Pre-Treatment-vs-Early", ksea_FDR = 0.05)

```

Below we collect the most significant differentially expressed genes using the comparisons stated above, 
as well as a KSEA plot made using PhosphoSite Plus + NetworKIN.
With a logarithmic scale we collect the adjusted p-values for each Kinase. 
Note the KSEA plot shows only those Kinases with at least 5 substrates found in the data, 
in addition to having an adjusted p-value below 0.05.

### Healthy vs Treated

We compare the healthy donor samples to all Treated samples.
Red indicates increased activity in treated samples, while blue shows increased activity
in healthy samples.

```{r Treated vs Healthy}
show <- phos.diffex.treated %>%
  select(-featureID, -t, -B) %>% 
  head(20)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Treated-ksea-plot.png")
```

### Healthy vs Pre-Treatment

Next, we compare healthy samples to Pre-Treatment samples.
Red indicates increased activity in Pre-Treatment samples, while blue shows increased activity
in healthy samples.

```{r Pre-Treatment vs Healthy}
show <- phos.diffex.pre %>%
  select(-featureID, -t, -B) %>% 
  head()
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Pre-Treatment-ksea-plot.png")
```

### Healthy vs Treated Early

Next, we compare healthy samples to Treated early samples.
Red indicates increased activity in Treated early samples, while blue shows increased activity
in healthy samples.

```{r Treated early vs Healthy}
show <- phos.diffex.early %>%
  select(-featureID, -t, -B) %>% 
  head(20)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Treated, early-period-ksea-plot.png")
```

### Healthy vs Treated Late

Next, we compare healthy samples to Treated late samples.
Red indicates increased activity in Treated late samples, while blue shows increased activity
in healthy samples.

```{r Treated late vs Healthy}
show <- phos.diffex.late %>%
  select(-featureID, -t, -B) %>% 
  head()
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Treated, late-period-ksea-plot.png")
```

### Pre-Treatment vs Treated Early

Finally, we compare Pre-Treatment samples to Treated early samples.
Red indicates increased activity in Treated early samples, while blue shows increased activity
in Pre-Treatment samples.

```{r Early vs Pre-Treatment}
show <- phos.diffex.earlyvspre %>%
  select(-featureID, -t, -B) %>% 
  head()
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedPre-Treatment-vs-Early-ksea-plot.png")
```
