---
title: "Sunil Ex17 Global differential expression & GSEA"
author: "Camilo Posso"
date: "06/01/2021"
output: 
  html_document:
    code_folding: hide
    toc: true
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r libraries, include=FALSE}
library(amlresistancenetworks)
library(kableExtra)
library(dplyr)
library(readr)
source("../../Util/synapseUtil.R")

# corrected for loading mass and plexID, long time and subtle
g.gene.corrected <- querySynapseTable("syn25706561")

prot.dat <- g.gene.corrected

# from here we can select just the metadata of interest
metadata.columns = c('Sample','BeatAML Patient ID','Plex','Loading Mass', 'Description')
summary <- prot.dat%>%
  select(metadata.columns)%>%
  distinct()
summary$Description <- as.character(summary$Description)
summary$Type <- case_when(summary$`BeatAML Patient ID` == "Healthy Donor Stroma" ~ "Healthy Donor Stroma",
                          summary$`BeatAML Patient ID` == "Cell line" ~ "Cell line")
summary[is.na(summary$Type), "Type"] <- "Treated"

## Elie has corrected the time period annotation. Adding a pre-treatment, early, and late period.
correct.time.annotation <- as.data.frame(read_csv("../Updated time period annotation.csv"))
summary$Period <- correct.time.annotation[summary$Sample, "Classification"]
summary$Period <- case_when(grepl("early", summary$Period) ~ "Early",
                            grepl("Late", summary$Period) ~ "Late",
                            grepl("pre-treatment", summary$Period) ~ "Pre-Treatment")
```

### Get dataset

We access synapse to load the data in long form. We use data which has been corrected for both
Loading mass effect, as well as plex effect. Below we see a metadata table for this experiment.

```{r pressure, echo=FALSE}
show <- summary %>%
  select(-`BeatAML Patient ID`)
kbl(show) %>%
  kable_paper("hover", full_width = F)

```

In this report, we collect the results of rank based GSEA, as well as differential expression. We perform
5 comparisons overall. First, we compare the Healthy group to the Treated group. Now, as we can see,
the Treated group itself consists of 3 subgroups determined by the Period of treatment, namely Pre-Treatment,
Early and Late, so we compare the Healthy group to each of these 3 subgroups as well. Finally, we compare the
Pre-Treatment group directly to the Early group.

```{r prepare groups, limma, and early vs late gsea, include=FALSE}
##select the samples of interest
healthy <- subset(summary, Type == 'Healthy Donor Stroma')%>%
  select(Sample)%>%
  unique()

treated.late <- subset(summary, Period == "Late")%>%
  select(Sample)%>%
  unique()

treated.early <- subset(summary, Period == "Early")%>%
  select(Sample)%>%
  unique()

treated.pre <- subset(summary, Period == "Pre-Treatment") %>%
  select(Sample) %>%
  unique()

treated <- subset(summary, Type == 'Treated')%>%
  select(Sample)%>%
  unique()

##then we spread the proteomics data into a matrix
colnames(prot.dat)[[1]] <- "Gene"
prot.mat <- prot.dat%>%
  select(LogRatio,Sample,Gene)%>%
  tidyr::pivot_wider(values_from='LogRatio',names_from='Sample',
                      values_fn=list(LogRatio=mean),values_fill=0.0)%>%
  tibble::column_to_rownames('Gene')

## LogFC computed as treated - healthy.
prot.diffex.treated <- limmaTwoFactorDEAnalysis(prot.mat, healthy$Sample, treated$Sample) %>%
  rename(value = "logFC") %>%
  mutate(Gene = rownames(.))

## LogFC computed as pre-treatment - healthy
prot.diffex.pre <- limmaTwoFactorDEAnalysis(prot.mat, healthy$Sample, treated.pre$Sample) %>%
  rename(value = "logFC") %>%
  mutate(Gene = rownames(.))

## LogFC computed as early - healthy
prot.diffex.early <- limmaTwoFactorDEAnalysis(prot.mat, healthy$Sample, treated.early$Sample) %>%
  rename(value = "logFC") %>%
  mutate(Gene = rownames(.))

## LogFC computed as late - healthy
prot.diffex.late <- limmaTwoFactorDEAnalysis(prot.mat, healthy$Sample, treated.late$Sample) %>%
  rename(value = "logFC") %>%
  mutate(Gene = rownames(.))

## LogFC computed as early - pre-treatment
prot.diffex.earlyvspre <- limmaTwoFactorDEAnalysis(prot.mat, treated.pre$Sample, treated.early$Sample) %>%
  rename(value = "logFC") %>%
  mutate(Gene = rownames(.))

## LogFC computed as late - early
prot.diffex.latevsearly <- limmaTwoFactorDEAnalysis(prot.mat, treated.early$Sample, treated.late$Sample) %>%
  rename(value = "logFC") %>%
  mutate(Gene = rownames(.))

############################ Saving tables for further use ##################################
# prot.diffex.treated <- prot.diffex.treated %>%
#   rename(logFC = value) %>%
#   write.table("Treated - Healthy.txt", sep = "\t")
# 
# prot.diffex.pre <- prot.diffex.pre %>%
#   rename(logFC = value) %>%
#   write.table("Pre-Treatment - Healthy.txt", sep = "\t")
# 
# prot.diffex.early <- prot.diffex.early %>%
#   rename(logFC = value) %>%
#   write.table("Early - Healthy.txt", sep = "\t")
# 
# prot.diffex.late <- prot.diffex.late %>%
#   rename(logFC = value) %>%
#   write.table("Late - Healthy.txt", sep = "\t")
# 
# prot.diffex.earlyvspre <- prot.diffex.earlyvspre %>%
#   rename(logFC = value) %>%
#   write.table("Early - Pre-Treatment.txt", sep = "\t")
# 
# prot.diffex.latevsearly <- prot.diffex.latevsearly %>%
#   rename(logFC = value) %>%
#   write.table("Late - Early.txt", sep = "\t")

```


```{r computing GSEA, eval=FALSE, include=FALSE}
## Run this to make the pictures, saved at script location
set.seed(117)

plotOldGSEA(prot.diffex.treated, "Healthy-vs-Treated", width = 15)

plotOldGSEA(prot.diffex.pre, "Healthy-vs-Pre-Treatment", width = 15)

plotOldGSEA(prot.diffex.early, "Healthy-vs-Treated, early-period", width = 15)

plotOldGSEA(prot.diffex.late, "Healthy-vs-Treated, late-period", width = 15)

plotOldGSEA(prot.diffex.earlyvspre, "Pre-Treatment-vs-Early", width = 15)

plotOldGSEA(prot.diffex.latevsearly, "Early-vs-Late", width = 15)

```

Below we collect the most significant differentially expressed genes using the comparisons stated above, 
as well as a GSEA plot made using a ranking of log-fold changes in gene expression.
With a logarithmic scale we collect the adjusted p-values of each normalized enrichment score. 
Note the GSEA plot shows only the top 20 most enriched gene sets among those 
with an adjusted p-value below 0.05.

### Healthy vs Treated

We compare the healthy donor samples to all Treated samples.
Red indicates increased activity in treated samples, while blue shows increased activity
in healthy samples.

```{r Treated vs Healthy}
show <- prot.diffex.treated %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Treated-gseaGO-plot.png")
```

### Healthy vs Pre-Treatment

Next, we compare healthy samples to Pre-Treatment samples.
Red indicates increased activity in Pre-Treatment samples, while blue shows increased activity
in healthy samples.

```{r Pre-Treatment vs Healthy}
show <- prot.diffex.pre %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Pre-Treatment-gseaGO-plot.png")
```

### Healthy vs Treated Early

Next, we compare healthy samples to Treated early samples.
Red indicates increased activity in Treated early samples, while blue shows increased activity
in healthy samples.

```{r Treated early vs Healthy}
show <- prot.diffex.early %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Treated, early-period-gseaGO-plot.png")
```

### Healthy vs Treated Late

Next, we compare healthy samples to Treated late samples.
Red indicates increased activity in Treated late samples, while blue shows increased activity
in healthy samples.

```{r Treated late vs Healthy}
show <- prot.diffex.late %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedHealthy-vs-Treated, late-period-gseaGO-plot.png")
```

### Pre-Treatment vs Treated Early

Finally, we compare Pre-Treatment samples to Treated early samples.
Red indicates increased activity in Treated early samples, while blue shows increased activity
in Pre-Treatment samples.

```{r Early vs Pre-Treatment}
show <- prot.diffex.earlyvspre %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedPre-Treatment-vs-Early-gseaGO-plot.png")
```

### Early vs Late

Finally, we compare the treated Early samples to the treated Late samples.
Red indicates increased activity in Treated late samples, while blue shows increased activity
in Pre-Treatment samples.

```{r Late vs Early}
show <- prot.diffex.latevsearly %>%
  select(-featureID, -t, -B) %>% head(.)
kbl(show) %>%
  kable_paper("hover", full_width = F)
```

```{r, include=TRUE, fig.align="center"}
knitr::include_graphics("sig-includedEarly-vs-Late-gseaGO-plot.png")
```

