---
title: "Global proteomics pipeline for Ex17 using PlexedPiper"
author: "Michael Nestor"
date: "4/16/2021"
output: rmarkdown::github_document
---

# Global Proteomics analysis

The global proteomics pipeline uses the R package [PlexedPiper](https://github.com/vladpetyuk/PlexedPiper). It also requires a connection to the DMS to access data packages.

```{r, echo=F}
library(knitr)
knitr::opts_chunk$set(message=F, warning=F)
```

```{r}
library(PlexedPiper)
data_package_num <- 3813
path_to_data_folder <- "data/"
if (!is_PNNL_DMS_connection_successful()) {
  stop("No connection to DMS.")
}
```

## 1. Read study design information

Study design information in PlexedPiper is encoded in three tables: fractions, samples, and references. These tables can be made using metadata and should be stored on the DMS before processing.

```{r}
study_design <- get_study_design_by_dataset_package(data_package_num)

fractions <- study_design$fractions
samples <- study_design$samples
references <- study_design$references
```

## 2 Processing MS-GF+ data

MS-GF+ data is processed in several steps. First, read MS-GF+ output from the DMS. (This step can take a while).

```{r}
msgf_data_path <- "data/msgfData_original.RData"

if (file.exists(msgf_data_path)) {
  load(msgf_data_path)
} else {
  msnid <- read_msms_data_from_DMS(data_package_num)
  save(msnid, file=msgf_data_path)
}

show(msnid)
```

### 2.1 Remap accessions

This function remaps UniProt protein accessions to gene symbol. First, we require a conversion table with two columns, `UNIPROT` and `SYMBOL`.

```{r}
path_to_FASTA <- path_to_FASTA_used_by_DMS(data_package_num)

library(Biostrings)
fst <- readAAStringSet(path_to_FASTA)

library(dplyr)
library(stringr)
conv <- data.frame(NAME = names(fst)) %>%
  mutate(UNIPROT = str_extract(names(fst), "^(\\S)+"),
         SYMBOL = str_extract(names(fst), "GN=(\\S)+")) %>%
  dplyr::select(-NAME) %>%
  mutate(UNIPROT = sub(".*\\|(.*)\\|.*", "\\1", UNIPROT),
         SYMBOL = sub("GN=", "", SYMBOL))

head(conv)
```

```{r}
msnid <- remap_accessions_uniprot_to_gene(msnid,
                                          organism_name="Homo sapiens",
                                          conversion_table=conv)
table(is.na(msnid$accession))
```

```{r}
msnid <- apply_filter(msnid, "!is.na(accession)")
show(msnid)
```

### 2.2 FDR filter

We use the target-decoy search strategy method described in [(Elias 2010)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2922680/). Filtering is done first at peptide level, then at protein level, both with max FDR of 1%.

```{r}
msnid <- filter_msgf_data(msnid, level="peptide", fdr.max=0.01)
show(msnid)
```

```{r}
path_to_FASTA <- path_to_FASTA_used_by_DMS(data_package_num)
path_to_FASTA <- gsub("\\\\", "/", path_to_FASTA)
file.copy(path_to_FASTA, path_to_data_folder)
path_to_FASTA_gene <- remap_accessions_uniprot_to_gene_fasta(path_to_FASTA)
```

```{r}
msnid <- compute_num_peptides_per_1000aa(msnid, path_to_FASTA_gene)
msnid <- filter_msgf_data(msnid, level="accession", fdr.max=0.01)
show(msnid)
```

```{r}
msnid <- apply_filter(msnid, "!isDecoy")
show(msnid)
```


### 2.3 Parsimonious inference

To reduce number of protein identifications, we use a parsimonious inference algorithm described in [(Zhang et al. 2007)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2810678/).

```{r}
msnid <- infer_parsimonious_accessions(msnid, unique_only=TRUE)
show(msnid)
```

```{r}
save(msnid, file=file.path(path_to_data_folder, "msgfData_filtered.RData"))
```

## 3 Process MASIC data

Output from the MASIC software is read from DMS, then filtered by inteference score.

```{r}
masic_data_path <- "data/masicData_original.RData"

if (file.exists(masic_data_path)) {
  load(masic_data_path)
} else {
  masic_data <- read_masic_data_from_DMS(data_package_num,
                                       interference_score = TRUE)
  save(msnid, file=masic_data_path)
}

nrow(masic_data)
```
```{r}
masic_data <- filter_masic_data(masic_data,
                                interference_score_threshold = 0.5,
                                s2n_threshold = 0)
save(masic_data, file=file.path(path_to_data_folder, "masicData_filtered.RData"))

nrow(masic_data)
```

## 4 Create crosstab

The quantitative crosstab combines MS/MS identifications with reporter ion intensities. Abundances are taken relative to the reference channel and then log-transformed.

```{r}
aggregation_level <- c("accession")
crosstab <- create_crosstab(msnid, masic_data,
                            aggregation_level,
                            fractions, samples, references)
dim(crosstab)
```

```{r}
library(MSnbase)
load("data/phenoData.RData")
m <- MSnSet(as.matrix(crosstab))

pData(m) <- p[sampleNames(m),]

save(m, file="data/msnset_global_gene_original.RData")
```

```{r}
aggregation_level <- c("accession", "peptide")
crosstab <- create_crosstab(msnid, masic_data,
                            aggregation_level,
                            fractions, samples, references)
dim(crosstab)
```

```{r}
m <- MSnSet(as.matrix(crosstab))

pData(m) <- p[sampleNames(m),]

save(m, file="data/msnset_global_peptide_original.RData")
```